src/connector/src/macros.rs

```diff
@@ -315,6 +315,57 @@ macro_rules! impl_connector_properties {
                 }
             }
         )*
+
+
+        $crate::paste! {
+            impl ConnectorProperties {
+                pub async fn create_split_enumerator(self, context: $crate::source::base::SourceEnumeratorContextRef) -> $crate::error::ConnectorResult<Box<dyn AnySplitEnumerator>> {
```

Comment:
Can we move the method signatures out of the macro but only generate the bodies for better DX?

Reply:
can, but why does it have better DX? :eyes:

Reply:
Oh, you mean move out of `impl_connector_properties`, and call `dispatch_source_prop` instead?